depot/app/controllers/admin/base_controller.rb

```diff
@@ -0,0 +1,10 @@
+class Admin::BaseController < ApplicationController
+  before_action :ensure_user_is_admin
+
+  protected
+    def ensure_user_is_admin
+      unless current_user.role.eql? 'admin'
```

Comment:
create a current_admin method in application controller. that will return admin only.

So, if the user is not admin then it will give false